src/WebJobs.Extensions.DurableTask/Listener/TaskHttpActivityShim.cs

```diff
@@ -0,0 +1,112 @@
+            return serializedOutput;
+        }
+
+        private static async Task<DurableHttpResponse> CreateDurableHttpResponseAsync(HttpResponseMessage response)
+        {
+            DurableHttpResponse durableHttpResponse = new DurableHttpResponse(response.StatusCode);
+            durableHttpResponse.Headers = CreateStringValuesHeaderDictionary(response.Headers);
+            durableHttpResponse.Content = await response.Content.ReadAsStringAsync();
+            return durableHttpResponse;
+        }
+
+        private static IDictionary<string, StringValues> CreateStringValuesHeaderDictionary(IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers)
+        {
+            IDictionary<string, StringValues> newHeaders = new Dictionary<string, StringValues>();
+            if (headers != null)
+            {
+                foreach (var header in headers)
+                {
+                    foreach (var headerValue in header.Value)
+                    {
+                        if (newHeaders.ContainsKey(header.Key))
+                        {
+                            StringValues values = StringValues.Concat(headerValue, newHeaders[header.Key]);
+                            newHeaders[header.Key] = values;
+                        }
+                        else
+                        {
+                            StringValues stringValues = new StringValues(headerValue);
+                            newHeaders.Add(header.Key, stringValues);
+                        }
+                    }
+                }
+            }
+
+            return newHeaders;
+        }
+
+        private static async Task<HttpRequestMessage> CreateHttpRequestMessageAsync(string rawInput)
+        {
+            JsonSerializerSettings serializerSettings = new JsonSerializerSettings();
+            serializerSettings.TypeNameHandling = TypeNameHandling.Auto;
+
+            IList<string> input = JsonConvert.DeserializeObject<List<string>>(rawInput, serializerSettings);
+            string durableHttpRequestString = input.First();
+            DurableHttpRequest durableHttpRequest = JsonConvert.DeserializeObject<DurableHttpRequest>(durableHttpRequestString, serializerSettings);
+
+            // Creating HttpRequestMessage
+            HttpRequestMessage requestMessage = new HttpRequestMessage(durableHttpRequest.Method, durableHttpRequest.Uri);
+            if (durableHttpRequest.Headers != null)
+            {
+                foreach (KeyValuePair<string, StringValues> entry in durableHttpRequest.Headers)
+                {
+                    requestMessage.Headers.Add(entry.Key, (IEnumerable<string>)entry.Value);
+                }
+            }
+
+            if (durableHttpRequest.Content != null)
+            {
+                requestMessage.Content = JsonConvert.DeserializeObject<HttpContent>(durableHttpRequest.Content);
```

Comment:
We should consider testing various types of HttpContent here, so we know what this does and doesn't work with.